main/src/ca/uwaterloo/flix/language/phase/Safety.scala

```diff
@@ -636,16 +740,16 @@ object Safety {
         case Denotation.Relational => terms
         case Denotation.Latticenal => terms.dropRight(1)
       }
-      val err2 = relTerms.flatMap(freeVarsOf).filter(latVars.contains).map(
-        s => IllegalRelationalUseOfLatticeVariable(s, loc)
-      )
+      val err2 = mapN(traverse(relTerms.flatMap(freeVarsOf).filter(latVars.contains)) {
+        s => IllegalRelationalUseOfLatticeVariable(s, loc).toFailure
+      })(_ => den)
```

Comment:
Why do we map to `den` here?

Reply:
I did that since `relTerms` is derived from `den`. This is mostly a result of me following the original structure somewhat, so I am not gonna argue it couldn't be cleaner.